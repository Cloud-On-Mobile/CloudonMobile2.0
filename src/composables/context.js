import { inject } from 'vue'
import { Context } from '@/classes/Context'

const injectionKey = 'context'

export function createContext(initialState = {}) {
  const context = new Context(initialState)

  return { injectionKey, context }
}

export function useContext() {
  const context = inject(injectionKey)
  if (!context) throw Error('fatal error: Context not provided!')

  return context
}
