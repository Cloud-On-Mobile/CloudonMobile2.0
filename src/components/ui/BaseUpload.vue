<script lang="ts" setup>
import BaseButton from '@/components/ui/BaseButton.vue'
import { useContext } from '@/composables/context'
import File from '@/types/File'

const ctx = useContext()
const { webSocketService } = ctx
const MAX_SIZE_FILE_IN_BYTES = 104857600

function isFileValid(file: File) {
  return file.size < MAX_SIZE_FILE_IN_BYTES ? true : false
}

function handleFileChange(event: { target: { files: File[] } }) {

  if (event.target.files && event.target.files[0]) {
    if (isFileValid(event.target.files[0])) {
      webSocketService.sendFile(event.target.files[0])
    } else {
      console.log('Invalid file')
    }
  }
}
</script>

<template>
  <BaseButton
    class="dashboard-files__button"
    theme="new-file"
    @change="handleFileChange"
  >
    <label for="selectFile">+ Add new file</label>
    <input
      id="selectFile"
      ref="newFile"
      type="file"
      class="selectFiles"
    />
  </BaseButton>
</template>

<style lang="scss" scoped>
input {
  display: none;
}
label {
  padding: 26px calc(100% - 170px)
}
</style>
