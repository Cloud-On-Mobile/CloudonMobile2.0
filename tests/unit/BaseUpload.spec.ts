import { mount } from '@vue/test-utils'
import BaseUpload from '@/components/ui/BaseUpload.vue'
import { createContext } from '@/composables/context'

describe('BaseUpload', () => {
  it('renders correctly', () => {
    const { context } = createContext()
    const wrapper = mount(BaseUpload, {
      global: {
        provide: {
          context: context
        }
      }
    });

    // Assertions
    expect(wrapper.element).toMatchSnapshot()
  })
})
