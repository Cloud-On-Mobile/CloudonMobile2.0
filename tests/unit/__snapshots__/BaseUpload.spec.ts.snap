// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseUpload renders correctly 1`] = `
<button
  class="base-button base-button--new-file dashboard-files__button"
>
  
  <label
    for="selectFile"
  >
    + Add new file
  </label>
  <input
    class="selectFiles"
    id="selectFile"
    type="file"
  />
  
</button>
`;
